plugins/woocommerce/src/Internal/Admin/EmailPreview/EmailPreview.php

```diff
@@ -254,16 +254,42 @@ public function render() {
 	 * @return string
 	 */
 	public function ensure_links_open_in_new_tab( string $content ) {
-		return (string) preg_replace_callback(
-			'/<a\s+([^>]*?)(target=["\']?[^"\']*["\']?)?([^>]*)>/i',
-			function ( $matches ) {
-				$before = $matches[1];
-				$target = 'target="_blank"';
-				$after  = $matches[3];
-				return "<a $before $target $after>";
-			},
-			$content
-		);
+		if ( empty( $content ) || strpos( $content, '<a' ) === false ) {
+			return $content;
+		}
+
+		// Suppress libxml errors to prevent them from being displayed.
+		$previous_use_internal_errors = libxml_use_internal_errors( true );
+
+		try {
+			$dom = new \DOMDocument();
+
+			// Add UTF-8 encoding and load with error suppression flags.
+			$html_with_encoding = '<?xml encoding="UTF-8">' . $content;
+			$dom->loadHTML(
+				$html_with_encoding,
+				LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD | LIBXML_NOWARNING | LIBXML_NOERROR
+			);
+
+			$links = $dom->getElementsByTagName( 'a' );
+			foreach ( $links as $link ) {
+				$link->setAttribute( 'target', '_blank' );
+				$link->setAttribute( 'rel', 'noopener' );
+			}
+
+			$result = $dom->saveHTML();
+
+			// Remove the XML declaration and DOCTYPE.
```

Comment:
Why do we remove those? Do you think we should improve this comment to explain this better?